### PR TITLE
Refactoring the program

### PR DESCRIPTION
**Pull request by AnthonySteele**
Extracted `Registry` and `Runner` classes from the program.
The registry is the IoC config part.
The runner is the async part, and all it's args turn into constructor params, which can be found by using the container :)
Is naming ok? "Runner" is a fairly generic name.

### CODE REVIEW

File: NuKeeper/Registry.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using NuKeeper.Configuration;
+using NuKeeper.Engine;
+using NuKeeper.Github;
+using NuKeeper.NuGet.Api;
+using SimpleInjector;
+
+namespace NuKeeper
+{
+    public static class Registry
```
Comment (cohen990) [inline NuKeeper/Registry.cs]: ContainerRegistration.Init or IOC.RegisterContainer seems more appropriate - Registry makes me think of operating systems and RAM

File: NuKeeper/Runner.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            _github = github;
+            _githubToken = settings.GithubToken;
+        }
+
+        public async Task Run(string tempDir)
+        {
+            var githubUser = await _github.GetCurrentUser();
+            Console.WriteLine($"Read github user '{githubUser}'");
+
+            var git = new LibGit2SharpDriver(tempDir, githubUser, _githubToken);
```
Comment (cohen990) [inline NuKeeper/Runner.cs]: can we modify things so this is injectable?

File: NuKeeper/Program.cs [hunk 3]
```diff
@@ -23,70 +23,16 @@ public static int Main(string[] args)
...
             var tempDir = TempFiles.MakeUniqueTemporaryPath();
 
-            RunAll(container.GetInstance<IGithubRepositoryDiscovery>(),
-                container.GetInstance<IPackageUpdatesLookup>(),
-                container.GetInstance<IPackageUpdateSelection>(),
-                container.GetInstance<IGithub>(),
-                tempDir,
-                container.GetInstance<Settings>().GithubToken)
+            var runner = container.GetInstance<Runner>();
+            runner.Run(tempDir)
```
Comment (cohen990) [inline NuKeeper/Program.cs]: I always liked Engine.Run(...) just for aesthetic reasons :P

File: NuKeeper/Runner.cs (hunk 2 above)
Comment (AnthonySteele) [inline NuKeeper/Runner.cs]: I expect that most or all  of this will become injectable, but in later PRs

**Pull request merged**